libs/storage/include/storage/file_object.hpp

```diff
@@ -55,14 +55,7 @@ struct FileBlockType
   static_assert(BS > META_DATA_BYTES,
                 "Block size needs to exceed the min requirement for metadata");
 
-  FileBlockType()
-  {
-    // Ensures that padded bytes are not uninitialised.
-    memset(this, 0, sizeof(decltype(*this)));
```

Comment:
I have a vague suspicion that something in the code relied on the padding of the class being zeroed. Maybe we can get around this issue if we mark these structures as packed?

Reply:
In other words, something relied on UB.
> Maybe we can get around this issue if we mark these structures as packed?
Yes, a good idea.
Another way is — once we remove non-trivial ctors, technically, these structs turn into POD. Even if zeroing them is absolutely required, it shouldn't be done in their ctors (which breaks trivialness) but rather in an external function, so they are not instantiated directly anymore.

Reply:
Yeah I think packing would work @ejfitzgerald the only thing I am aware of like that would be if you have padding that you do not zero and you write it to a file, undefined behaviour sanitizer and/or address san complain (I vaguely recall). Also, why would you want random garbage in your file

Reply:
The hardest part is actually the physical storage layer where all the IO is performed via whole objects simply being sent as is to `std::fstream::read/write()` and `std::memcpy()`. This would work with objects data but unfortunately not every object kept in stacks can be called such, even if packed.
I have a plan for a better temporary fix which this PR is too narrow to contain. :\

Reply:
I am curious, what is your solution?